src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
The point of TryDoWithPlugin would be to move the logic of cLuaWorldTask into the lambda. eg: 
```
self->QueueTask(
    [&Resettable, =FnRef] (cWorld & a_World)
    {
        Resettable->TryDoWithPlugin([&a_World, =FnRef] (cPlugin & a_Plugin)
        {
             a_Plugin->Call(FnRef, a_World);
        })
    });
```
But maybe this is too far down the road to callback hell.